### ISSUE TITLE
1946-intra-rustdoc-links issue

### ISSUE THREAD

**Issue by olegnn** [rust-lang/rust#64193]
**Problem**

Links created according to [1946-intra-rustdoc-links](https://rust-lang.github.io/rfcs/1946-intra-rustdoc-links.html) either don't work as expected or work with ```warning: `[...]` cannot be resolved, ignoring it...```.

**Steps**

1. Go to https://rust-lang-nursery.github.io/futures-api-docs/0.3.0-alpha.18/futures/channel/mpsc/struct.TryRecvError.html.
2. Click on `try_next` link.
It won't work correctly, but if we
3. ```git clone https://github.com/rust-lang-nursery/futures-rs```
4. ```cd futures-rs```
5. Change [line](https://github.com/rust-lang-nursery/futures-rs/blob/master/futures-channel/src/mpsc/mod.rs#L165)
from
```[`try_next`](Receiver::try_next)```
to
```[`try_next`](mpsc::Receiver::try_next())```
or to
```[`try_next`](futures_channel::mpsc::Receiver::try_next())```
6. After ```cargo +nightly doc```
we will get
```warning: `[mpsc::Receiver::try_next]` cannot be resolved, ignoring it...```
or
```warning: `[futures_channel::mpsc::Receiver::try_next]` cannot be resolved, ignoring it...```
respectively.

__Meanwhile link will magically work.__

**Possible Solution(s)**

It would be nice to remove incorrect warnings and make doc example work [or at least add warning for that case]. Thanks.

**Notes**

Output of `cargo version`:

```cargo 1.39.0-nightly (22f7dd049 2019-08-27)```

**Comment by ehuss** [comment]
Transferred to rust-lang/rust since this is a rustdoc issue.

**Comment by lzutao** [comment]
Link to the tracking issue of intra doc link: #43466

**Comment by jyn514** [comment]
Possible duplicate of https://github.com/rust-lang/rust/issues/73699 (I know this issue came first but the other has a MCVE).

**Comment by jyn514** [comment]
Closing as duplicate of https://github.com/rust-lang/rust/issues/73699: `futures_rs` is trying to link to the upstream crate `futures_channel`.

**Comment by Nemo157** [comment]
This isn't related to #73699/#74481, it's just normal cross-crate inlining intradoc issues.

On [`futures-channel::mpsc::TryRecvError`](https://docs.rs/futures-channel/0.3.5/futures_channel/mpsc/struct.TryRecvError.html) the link is correctly resolved, on [`futures::mpsc::TryRecvError`](https://docs.rs/futures/0.3.5/futures/channel/mpsc/struct.TryRecvError.html) it is not. I tried building with the latest nightly (`rustdoc 1.46.0-nightly (346aec9b0 2020-07-11)`) and it still doesn't work, @jyn514 are there still known cross-crate inlining issues, or should they all be fixed now?

**Comment by Nemo157** [comment]
Testing with _actual_ latest nightly works :tada: so this will be fixed for the next version published.
